src/stores/video.ts

```diff
@@ -495,31 +498,146 @@ export const useVideoStore = defineStore('video', () => {
     })
   })
 
-  // Process videos that were being recorded when the app was closed
-  const processUnprocessedVideos = async (): Promise<void> => {
-    if (keysFailedUnprocessedVideos.value.isEmpty()) return
-    console.log(`Processing unprocessed videos: ${keysFailedUnprocessedVideos.value.join(', ')}`)
-
-    const chunks = await tempVideoChunksDB.keys()
-    if (chunks.length === 0) {
-      discardUnprocessedVideos()
-      throw new Error('No video recording data found. Discarding leftover info.')
+  const updateLastProcessingUpdate = (recordingHash: string): void => {
+    const info = unprocessedVideos.value[recordingHash]
+    info.dateLastProcessignUpdate = new Date()
+    unprocessedVideos.value = {
+      ...unprocessedVideos.value,
+      ...{ [recordingHash]: info },
     }
+  }
 
-    const processingErrors: string[] = []
-    for (const recordingHash of keysFailedUnprocessedVideos.value) {
-      const info = unprocessedVideos.value[recordingHash]
-      console.log(`Processing unprocessed video: ${info.fileName}`)
+  // Progress tracking for video processing
+  const videoProcessingDetails = ref<VideoProcessingDetails>({})
+  const totalFilesToProcess = ref(0)
+
+  const updateFileProgress = (filename: string, progress: number, message: string): void => {
+    videoProcessingDetails.value[filename] = { filename, progress, message }
+  }
+
+  const currentFileProgress = computed(() => {
+    return Object.values(videoProcessingDetails.value).map((detail) => ({
+      fileName: detail.filename,
+      progress: detail.progress,
+      message: detail.message,
+    }))
+  })
+
+  const overallProgress = computed(() => {
+    const entries = Object.values(videoProcessingDetails.value)
+    const totalProgress = entries.reduce((acc, curr) => acc + curr.progress, 0)
+    return (totalProgress / (totalFilesToProcess.value * 100)) * 100
+  })
+
+  const processVideoChunksAndTelemetry = async (hashes: string[]): Promise<void> => {
+    totalFilesToProcess.value = hashes.length
+    videoProcessingDetails.value = {}
+
+    const debouncedUpdateFileProgress = useDebounceFn((filename: string, progress: number, message: string) => {
+      updateFileProgress(filename, progress, message)
+    }, 100)
+
+    const videosToProcess: UnprocessedVideoInfo[] = hashes.map((hash) => unprocessedVideos.value[hash])
```

Comment:
By doing an `Object.entries` here, you can avoid reassigning the object to an array and the need to add the hash as a property to the `UnprocessedVideo` type.

There's also the possibility of keeping the new property, but maybe in this case we should not use the hash as a key and directly use an array for the `unprocessedVideos` directly?